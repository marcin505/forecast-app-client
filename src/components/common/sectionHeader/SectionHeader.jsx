import React from 'react';
import PropTypes from 'prop-types';
import CloseContainer from 'components/common/closeContainer/CloseContainer.jsx';
import './SectionHeader.css';

const SectionHeader = ({closeAction, heading}) => {
   return (
      <div className="section-header">
         <div className="heading-red-sm">{heading}</div>
         {
            <CloseContainer
               closeAction={closeAction}
               className='closeContainer'
            />
         }
      </div>
   );
};

SectionHeader.propTypes = {
   closeAction: PropTypes.func.isRequired,
   heading: PropTypes.string.isRequired,
};

export default SectionHeader;