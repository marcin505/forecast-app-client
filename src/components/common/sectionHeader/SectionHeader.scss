.section-header {
  display: flex;
  margin-bottom: 30px;
}
