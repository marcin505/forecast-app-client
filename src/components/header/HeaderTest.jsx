import React, { Component } from 'react'
import PropTypes from 'prop-types';

const TodoTextInput = () => (
   <div className="footer content-wrapper">
      <p>
         Copyright Copyright
      </p>
   </div>
);
class HeaderTest extends Component {
   handleSave(text) {
      if (text.length !== 0) {
         this.props.addTodo(text)
      }
   }

   render() {
      return (
         <header className="header">
            <h1>todos</h1>
            <TodoTextInput
               newTodo={true}
               onSave={this.handleSave.bind(this)}
               placeholder="What needs to be done?"
            />
         </header>
      )
   }
}

HeaderTest.propTypes = {
   addTodo: PropTypes.func.isRequired
};

export default HeaderTest;