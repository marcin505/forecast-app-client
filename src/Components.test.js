import React from 'react';
import Enzyme, {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});
import HeaderTest from 'components/header/HeaderTest';
import SectionHeader from 'components/common/sectionHeader/SectionHeader';

describe('components', () => {
   let enzymeWrapper;
   describe('>>>HeaderTest --- Shallow Render HeaderTest', () => {
      const props = {
         addTodo: jest.fn()
      }
      beforeEach(() => {
         enzymeWrapper = shallow(<HeaderTest {...props}/>)
      });
      it('should render self and subcomponents', () => {
         expect(enzymeWrapper.length).toEqual(1);
         expect(enzymeWrapper.find('header').hasClass('header')).toBe(true);
         const todoInputProps = enzymeWrapper.find('TodoTextInput').props();
         expect(todoInputProps.placeholder).toEqual('What needs to be done?');
      });
      it('should call addTodo if length of text is greater than 0', () => {
         const input = enzymeWrapper.find('TodoTextInput');
         input.props().onSave('')
         expect(props.addTodo.mock.calls.length).toBe(0);
         input.props().onSave('Use Redux')
         expect(props.addTodo.mock.calls.length).toBe(1);
      });
   });
   describe('>>>SectionHeader --- Shallow Render HeaderTest', () => {
      const props = {
         closeAction: jest.fn(),
         heading: 'someHeading',
      }
      beforeEach(() => {
         enzymeWrapper = shallow(<SectionHeader {...props}/>)
      });
      it('should render self and subcomponents', () => {
         expect(enzymeWrapper.length).toEqual(1);
         expect(enzymeWrapper.find('CloseContainer').hasClass('closeContainer')).toBe(true);
      });
      it('should call closeAction', () => {
         const CloseContainer = enzymeWrapper.find('CloseContainer');
         CloseContainer.props().closeAction();
         expect(props.closeAction.mock.calls.length).toBe(1);
      });
   });
   describe('>>>SectionHeader --- Snapshot',()=>{
      const props = {
         closeAction: jest.fn(),
         heading: 'someHeading',
      }
      it('+++capturing Snapshot of Home', () => {
         const renderedValue =  renderer.create(<SectionHeader {...props}/>).toJSON()
         expect(renderedValue).toMatchSnapshot();
      });
   });
})