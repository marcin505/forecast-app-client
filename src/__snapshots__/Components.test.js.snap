// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components >>>SectionHeader --- Snapshot +++capturing Snapshot of Home 1`] = `
<div
  className="section-header"
>
  <div
    className="heading-red-sm"
  >
    someHeading
  </div>
  <div
    className="close-container"
    onClick={[Function]}
  >
    <h2>
      Close
    </h2>
    <span
      className="close"
    />
  </div>
</div>
`;
